### PR TITLE
CRM-62 page size custom valid

### PR DESCRIPTION
**Pull request by SteelWro**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Commits are squashed
- [x] PR and commit messages follow the style guidelines

### CODE REVIEW

File: backend/src/main/java/com/crm/exception/validator/ValidSize.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import static com.crm.exception.ErrorDict.*;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Target({FIELD})
+@Retention(RUNTIME)
+@Documented
+@Constraint(validatedBy = {ValidSize.ValidSizeValidator.class})
+public @interface ValidSize {
```
Comment (Fellway) [inline backend/src/main/java/com/crm/exception/validator/ValidSize.java:20]: Trochę duplikacja kodu. W sumie ma to samo co ValidPage. Może lepiej będzie jakoś to inaczej nazwać  i zastosować do size i page jeden interfejs?

**Review by Fellway — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/crm/exception/validator/ValidSize.java (hunk 1 above)
Comment (SteelWro) [reply backend/src/main/java/com/crm/exception/validator/ValidSize.java:20]: W sumie pomyślałem o tym, zmienię

Comment (SteelWro) [reply backend/src/main/java/com/crm/exception/validator/ValidSize.java]: done

File: backend/src/main/java/com/crm/App.java [hunk 2]
```diff
@@ -2,15 +2,13 @@
...
 import org.springframework.boot.autoconfigure.SpringBootApplication;
-import org.springframework.boot.context.ApplicationPidFileWriter;
 import springfox.documentation.swagger2.annotations.EnableSwagger2;
 
 @SpringBootApplication
 @EnableSwagger2
 public class App {
     public static void main(String[] args) {
         SpringApplication springApplication = new SpringApplication(App.class);
-        springApplication.addListeners(new ApplicationPidFileWriter());
```
Comment (aliwocha) [inline backend/src/main/java/com/crm/App.java:13]: This already has been deleted in another task

**Review by aliwocha — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/crm/exception/validator/ValidParam.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+@Target({ElementType.METHOD, ElementType.FIELD, ElementType.ANNOTATION_TYPE})
+@Retention(RUNTIME)
+@Documented
+@Constraint(validatedBy = {ValidParam.ValidSizeValidator.class})
+public @interface ValidParam {
+    String emptyParamMessage() default "empty parameter";
+
+    String invalidParamMessage() default "invalid parameter";
+
+    String message() default "blank";
```
[the remark this answers is not in the document]
Comment (SteelWro) [reply backend/src/main/java/com/crm/exception/validator/ValidParam.java:23]: fields: message, groups, payload i required for this interface on app starting

File: backend/src/main/java/com/crm/App.java (hunk 2 above)
Comment (SteelWro) [reply backend/src/main/java/com/crm/App.java:13]: I started this task two weeks ago before special task for this was added, my fault. I discard this change

Comment (aliwocha) [reply backend/src/main/java/com/crm/App.java:13]: I guess you can leave like it is.

Comment (SteelWro) [reply backend/src/main/java/com/crm/App.java:13]: done

File: backend/src/main/java/com/crm/exception/validator/ValidParam.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        public boolean isValid(String value, ConstraintValidatorContext context) {
+            String message = "";
+            if (value == null) {
+                message = emptyParamMessage;
+            } else if (value.equals("")) {
+                message = emptyParamMessage;
+            } else if (!positiveNumbersPattern.matcher(value).matches()) {
+               message = invalidParamMessage;
+            }
+            if (!message.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (SteelWro) [reply backend/src/main/java/com/crm/exception/validator/ValidParam.java:55]: done

File: backend/src/main/java/com/crm/exception/validator/ValidParam.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    String invalidParamMessage() default "invalid parameter";
+
+    String message() default "blank";
+
+    Class<?>[] groups() default {};
+
+    Class<? extends Payload>[] payload() default {};
+
+    class ValidSizeValidator implements ConstraintValidator<ValidParam, String> {
+        private final Pattern positiveNumbersPattern = Pattern.compile("^[0-9]*$");
```
[the remark this answers is not in the document]
Comment (SteelWro) [reply backend/src/main/java/com/crm/exception/validator/ValidParam.java:33]: done

File: backend/src/main/java/com/crm/dto/request/PageRequest.java [hunk 6]
```diff
@@ -1,16 +1,25 @@
...
+import static com.crm.exception.ErrorDict.PAGINATED_INVALID_SIZE;
 
-@Getter
 @Setter
 public class PageRequest {
-    @Min(value = 0, message = ErrorDict.PAGINATED_INVALID_PAGE)
-    private Integer page = 0;
-    @Min(value = 1, message = ErrorDict.PAGINATED_INVALID_SIZE)
-    private Integer size = 20;
+    @ValidParam(emptyParamMessage = PAGINATED_EMPTY_PAGE, invalidParamMessage = PAGINATED_INVALID_PAGE)
```
[the remark this answers is not in the document]
Comment (SteelWro) [reply backend/src/main/java/com/crm/dto/request/PageRequest.java:13]: done

File: backend/src/main/java/com/crm/exception/validator/ValidParam.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.regex.Pattern;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Target({ElementType.METHOD, ElementType.FIELD, ElementType.ANNOTATION_TYPE})
+@Retention(RUNTIME)
+@Documented
+@Constraint(validatedBy = {ValidParam.ValidSizeValidator.class})
+public @interface ValidParam {
+    String emptyParamMessage() default "empty parameter";
```
[the remark this answers is not in the document]
Comment (SteelWro) [reply backend/src/main/java/com/crm/exception/validator/ValidParam.java:21]: done

File: backend/src/main/java/com/crm/exception/validator/ValidParam.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
+package com.crm.exception.validator;
+
+import javax.validation.*;
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.security.InvalidParameterException;
+import java.util.Arrays;
```
[the remark this answers is not in the document]
Comment (SteelWro) [reply backend/src/main/java/com/crm/exception/validator/ValidParam.java]: done

File: backend/src/test-integration/java/com/crm/api/v1/CustomerControllerIntegrationTest.java [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+import static org.hamcrest.Matchers.is;
+import static org.springframework.http.HttpStatus.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest(classes = App.class)
+@AutoConfigureMockMvc
+@WithMockUser
+public class CustomerControllerIntegrationTest {
```
[the remark this answers is not in the document]
Comment (SteelWro) [reply backend/src/test-integration/java/com/crm/api/v1/CustomerControllerIntegrationTest.java]: done

File: backend/src/test-integration/java/com/crm/api/v1/CustomerControllerIntegrationTest.java [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+    @Test
+    public void shouldResponseWithDefaultValuesWhenCallingCustomersEndpoint() throws Exception {
+        mvc.perform(get("/customers"))
+                .andExpect(status().is(OK.value()))
+                .andExpect(jsonPath("$.pageable.pageNumber", is(0)))
+                .andExpect(jsonPath("$.pageable.pageSize", is(20)));
+    }
+
+    @Test
+    public void shouldResponseEntityHasPageNumberEqualTo2WhenParamSizeEqualTo1() throws Exception {
```
[the remark this answers is not in the document]
Comment (SteelWro) [reply backend/src/test-integration/java/com/crm/api/v1/CustomerControllerIntegrationTest.java]: done

**Review by Fellway — APPROVED** [review]

**Review by aliwocha — APPROVED** [review]

**Pull request merged**